pkg/kubelet/volumes.go

```diff
@@ -268,18 +254,21 @@ func (kl *Kubelet) newVolumeMounterFromPlugins(spec *volume.Spec, pod *api.Pod,
 	if err != nil {
 		return nil, fmt.Errorf("can't use volume plugins for %s: %v", spec.Name(), err)
 	}
-	if plugin == nil {
-		// Not found but not an error
-		return nil, nil
-	}
 	physicalMounter, err := plugin.NewMounter(spec, pod, opts)
 	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate volume physicalMounter for %s: %v", spec.Name(), err)
+		return nil, fmt.Errorf("failed to instantiate mounter for volume: %s using plugin: %s with a root cause: %v", spec.Name(), plugin.Name(), err)
 	}
 	glog.V(10).Infof("Used volume plugin %q to mount %s", plugin.Name(), spec.Name())
 	return physicalMounter, nil
 }
 
+// newVolumeAttacherFromPlugins attempts to find a plugin from a volume spec
+// and then create an Attacher.
+// Returns:
+//  - an attacher if one exists
+//  - an error if no plugin was found for the volume
+//    or the attacher failed to instantiate
+//  - nil if there is no appropriate attacher for this volume
```

Comment:
This is kind of the exception to the whole sentence rule -- this is fine as is re: the specification of return values